### PR TITLE
add ability to partition by key

### PR DESCRIPTION
**Pull request by prad-a-RuntimeException**

### CODE REVIEW

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 1]
```diff
@@ -56,240 +58,245 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(BigQuerySinkTask.class);
+
+    private final BigQuery testBigQuery;
+    private SchemaRetriever schemaRetriever;
+    private BigQueryWriter bigQueryWriter;
+    private BigQuerySinkTaskConfig config;
+    private RecordConverter<Map<String, Object>> recordConverter;
+    private Map<String, TableId> topicsToBaseTableIds;
+    private boolean useMessageTimeDatePartitioning;
+    private boolean usePartitionByCustomKey;
```
Comment (revigreg) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: rename as useCustomKeyPartitioning to unify the naming format?

**Review by revigreg — COMMENTED** [review]
the code is fine, please let me know if you want to fix the name or not. 
If not, I'll just approve this code.

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 1 above)
Comment (prad-a-RuntimeException) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: Will do!!!

**Review by revigreg — DISMISSED** [review]

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkTaskConfig.java [hunk 2]
```diff
@@ -30,141 +30,157 @@
...
+            ConfigDef.Importance.MEDIUM;
+    private static final String BIGQUERY_RETRY_WAIT_DOC =
+            "The minimum amount of time, in milliseconds, to wait between BigQuery backend or quota "
+                    + "exceeded error retry attempts.";
+
+    public static final String BIGQUERY_MESSAGE_TIME_PARTITIONING_CONFIG =
+            "bigQueryMessageTimePartitioning";
+    private static final ConfigDef.Type BIGQUERY_MESSAGE_TIME_PARTITIONING_CONFIG_TYPE =
+            ConfigDef.Type.BOOLEAN;
+    public static final String BIGQUERY_MESSAGE_PARTITIONING_BY_KEY_CONFIG =
```
Comment (revigreg) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkTaskConfig.java]: rename the variable name also.

**Review by revigreg — APPROVED** [review]

**Pull request merged**